src/__tests__/integration/data/aftr/sampleContractSrc.js

```diff
@@ -0,0 +1,209 @@
+    }
+    balance = 0;
+    if (target in balances) {
+      balance = balances[target];
+    }
+  }
+
+  if (input.function === "transfer") {
+    const target2 = input.target;
+    const qty = input.qty;
+    const callerAddress = isArweaveAddress(caller);
+    const targetAddress = isArweaveAddress(target2);
+    if (!Number.isInteger(qty)) {
+      throw new ContractError('Invalid value for "qty". Must be an integer.');
+    }
+    if (!targetAddress) {
+      throw new ContractError("No target specified.");
+    }
+    if (qty <= 0 || callerAddress === targetAddress) {
+      throw new ContractError("Invalid token transfer.");
+    }
+    if (!(callerAddress in balances)) {
+      throw new ContractError("Caller doesn't own a balance in the Vehicle.");
+    }
+    if (balances[callerAddress] < qty) {
+      throw new ContractError(`Caller balance not high enough to send ${qty} token(s)!`);
+    }
+    if (SmartWeave.contract.id === target2) {
+      throw new ContractError("A vehicle token cannot be transferred to itself because it would add itself the balances object of the vehicle, thus changing the membership of the vehicle without a vote.");
+    }
+    if (state.ownership === "single" && callerAddress === state.creator && balances[callerAddress] - qty <= 0) {
+      throw new ContractError("Invalid transfer because the creator's balance would be 0.");
+    }
+    balances[callerAddress] -= qty;
+    if (targetAddress in balances) {
+      balances[targetAddress] += qty;
+    } else {
+      balances[targetAddress] = qty;
+    }
+  }
+  if (input.function === "mint") {
+    if (!input.qty) {
+      throw new ContractError("Missing qty.");
+    }
+    if (!(caller in state.balances)) {
+      balances[caller] = input.qty;
+    }
+  }
+  if (input.function === "deposit") {
+    if (!input.txID) {
+      throw new ContractError("The transaction is not valid.  Tokens were not transferred to the vehicle.");
+    }
+    if (!input.tokenId) {
+      throw new ContractError("No token supplied. Tokens were not transferred to the vehicle.");
+    }
+    if (input.tokenId === SmartWeave.contract.id) {
+      throw new ContractError("Deposit not allowed because you can't deposit an asset of itself.");
+    }
+    if (!input.qty || typeof +input.qty !== "number" || +input.qty <= 0) {
+      throw new ContractError("Qty is invalid.");
+    }
+    let lockLength = 0;
+    if (input.lockLength) {
+      lockLength = input.lockLength;
+    }
+    console.log('====== AFTR CONTRACT deposit FN - calling claim on: ', input.tokenId);
+    const transferResult = await SmartWeave.contracts.write(input.tokenId, {
+      function: "claim",
+      txID: input.txID,
+      qty: input.qty
+    });
+    console.log('====== AFTR CONTRACT deposit FN - calling claim result ', transferResult);
+    console.log('====== AFTR CONTRACT deposit FN - PST state', transferResult.state);
+
+    const tokenInfo = getTokenInfo(transferResult.state);
```

Comment:
@arcj0 , after making an internal write, the most current state of the callee contract (i.e. the one, that internal write has been done) is in the result of the internal write